src/map/Pins.js

```diff
@@ -35,43 +63,155 @@
+                        .attr("text-anchor", function(d) {
+                            if (d[2].iconTextAnchor === undefined) {
+                                return "middle";
+                            }
+                            if (d[2].iconTextAnchor) {
+                                return d[2].iconTextAnchor;
+                            }
+                        })
+                        .attr("dx", function(d) {
+                            if (!d[2].iconTextAnchor && d[2].xIconOffset) {
+                                return -(pathBBox.width - d3.select(this).node().getBBox().width - d[2].xIconOffset);                            
+                            }
+                        })
+                        .attr("dy", function(d) {
+                            var offset = d[2].yIconOffset || 0;
+                            return -(pathBBox.height - d3.select(this).node().getBBox().height - offset);
+                        })
+                    ;
+                }
+            }
+            // img
+            if (d[2] && d[2].imgPath) {
+                d3.select(this).append("svg:image")
+                    .attr("class","pin-image")
+                    .attr('width', d[2].imgWidth + "px")
+                    .attr('height', d[2].imgHeight + "px")
+                    .attr("xlink:href", d[2].imgPath)
+                ;
+            }
+        });
+
         this.pinsPaths.exit().remove();
     };
 
     Pins.prototype.layerZoomed = function (base) {
         Layer.prototype.layerZoomed.apply(this, arguments);
+        var context = this;
         this.pinsPaths
             .attr("transform", function (d) {
                 var pos = base.project(d[0], d[1]);
                 if (!pos) {
                     pos = [0, 0];
                 }
-                return "translate(" + pos[0] + ", " + pos[1] + ")scale(" + 0.5 + ")";
+
+                var pinScale = context.scale();
+                if (d[2] && d[2].pinScale) {
+                    pinScale = d[2].pinScale;
+                }
+                return "translate(" + pos[0] + ", " + pos[1] + ")scale(" + pinScale + ")";
             })
-        ;
+          
+            .each(function(d, i) {
+                if (d[2] && d[2].imgPath) {
+                    d3.select(this).select(".pin-image")
+                        .attr("transform", function(d) {
+                            var yOffset, xOffset;
+                            if (d[2].iconOffset) {
+                                xOffset = d[2].iconOffset[0];
+                                yOffset = d[2].iconOffset[1];
+                            } else {
+                                yOffset = d3.select(this).node().getBBox().height;
+                                xOffset = d3.select(this).node().getBBox().width / 2;
+                            }
+                            return "translate(" + (-xOffset) + "," + (-yOffset) + ")";
```

Comment:
This function is too expensive to perform on each zoom + pan - the "nodes" width and height should not change during the scaling, so could probably be pre-calculated on the update call?